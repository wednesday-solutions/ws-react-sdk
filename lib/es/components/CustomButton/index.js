import{_ as t,a as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/button/style";import e from"antd/lib/button";import"../_rollup-plugin-inject-process-env-1259e43d.js";import o from"react";import n from"styled-components";import{c as m}from"../colors-93a064d8.js";import{f as a}from"../fonts-ca6864a0.js";import{m as i}from"../media-ee9cbc51.js";var s,p=n(e)(s||(s=t(["\n  &.ant-btn-lg {\n    ",";\n    border-radius: 8px;\n    height: 3.5rem;\n    ","\n    ","\n    ",";\n    margin-bottom: 1rem;\n    border: 2px solid ",";\n  }\n"])),(t=>t.width&&"width: ".concat(t.width,"%;")),i.desktop.max("\n      height: 3rem;\n    "),i.largeMobile.max("\n      height: 2.5rem;\n    "),a.style.xRegularFW600(),m.green);function l(t){return o.createElement(p,r({"data-testid":"custom-button"},t))}l.propTypes={};export default l;
