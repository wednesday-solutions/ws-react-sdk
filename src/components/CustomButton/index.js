/**
 *
 * CustomButton
 *
 */

import React from 'react';
import { Button } from 'antd';
import styled from 'styled-components';
import { colors, fonts, media } from '@themes';

const CB = styled(Button)`
  &.ant-btn-lg {
    ${props => props.width && `width: ${props.width}%;`};
    border-radius: 8px;
    height: 3.5rem;
    ${media.desktop.max(`
      height: 3rem;
    `)}
    ${media.largeMobile.max(`
      height: 2.5rem;
    `)}
    ${fonts.style.xRegularFW600()};
    margin-bottom: 1rem;
    border: 2px solid ${colors.green};
  }
`;

function CustomButton(props) {
  return <CB data-testid={'custom-button'} {...props} />;
}

CustomButton.propTypes = {};

export default CustomButton;
