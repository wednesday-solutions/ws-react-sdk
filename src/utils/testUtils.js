import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
