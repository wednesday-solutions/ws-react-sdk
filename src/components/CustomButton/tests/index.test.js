/**
 *
 * Tests for CustomButton
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import CustomButton from '../index';

describe('<CustomButton />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomButton />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomButton component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomButton />);
    expect(getAllByTestId('custom-button').length).toBe(1);
  });
});
