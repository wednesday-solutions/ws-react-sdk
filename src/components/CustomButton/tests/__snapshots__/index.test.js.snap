// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> should render and match the snapshot 1`] = `
<body>
  <div>
    <button
      class="ant-btn sc-bdfBwQ cDCNSQ"
      data-testid="custom-button"
      type="button"
    />
  </div>
</body>
`;
